optimade/models/utils.py

```diff
@@ -1,8 +1,108 @@
+import inspect
+import warnings
 import re
+from enum import Enum
+from typing import Optional
+
+from pydantic import Field
+
+_PYDANTIC_FIELD_KWARGS = list(inspect.signature(Field).parameters.keys())
 
 __all__ = ("CHEMICAL_SYMBOLS", "EXTRA_SYMBOLS", "ATOMIC_NUMBERS", "SemanticVersion")
 
 
+class SupportLevel(Enum):
+    MUST = "must"
+    SHOULD = "should"
+    OPTIONAL = "optional"
+
+
+def StrictField(
+    *args,
+    description: str = None,
+    **kwargs,
+) -> Field:
+    """A wrapper around `pydantic.Field` that does the following:
+
+    - Forbids any "extra" keys that would be passed to `pydantic.Field`,
+      except those used elsewhere to modify the schema in-place,
+      e.g. "uniqueItems", "pattern" and those added by OptimadeField, e.g.
+      "unit", "queryable" and "sortable".
+    - Emits a warning when no description is provided.
+
+    Raises:
+        RuntimeError: if a Field is created with an unexpected key.
+
+    Returns:
+        the pydantic field
+
+    """
+
+    allowed_keys = [
+        "unit",
+        "pattern",
+        "uniqueItems",
+        "support",
+        "queryable",
+        "sortable",
+    ]
+    _banned = [k for k in kwargs if k not in set(_PYDANTIC_FIELD_KWARGS + allowed_keys)]
+
+    if _banned:
+        raise RuntimeError(
+            f"Not creating StrictField({args}, {kwargs}) with forbidden keywords {_banned}."
+        )
+
+    if description is not None:
+        kwargs["description"] = description
+
+    if description is None:
+        warnings.warn(
+            f"No description provided for StrictField specified by {args}, {kwargs}."
+        )
+
+    return Field(*args, **kwargs)
+
+
+def OptimadeField(
+    *args,
+    support: Optional[SupportLevel] = SupportLevel.MUST,
+    queryable: Optional[SupportLevel] = SupportLevel.MUST,
```

Comment:
Should these be optional? I.e., shouldn't the type simply be `SupportLevel`?

Reply:
Good Q: I kept them optional as I imagine anyone using these for their own optimade models wouldn't want to specify a support level, but I guess they could just use `StrictField` for that now. Thoughts?

Reply:
Yeah, depending if they want a description or not, I guess, they would either use `Field`, `StrictField` or go to the extreme and use `OptimadeField`.

We should document this.

Reply:
For now, let's just keep it like you have - a bit more loose. And I think in the end `None`/`null` is perhaps a valid value here, no?

Reply:
`None` is valid *here*, but it doesn't get passed through to the creation of the field itself unless it has a `SupportLevel`. Again, you could argue either way here, but I'd prefer omission rather than `None` in the field itself

Reply:
But an omission would lead to a `MUST` level in both support and queryable, something one maybe wouldn't expect. But if you don't want `None`, then it may be prudent to check whether they are `None` and raise if `True`?

Reply:
Hmm, fair point. I thought I'd changed the default value to `None` in `OptimadeField` too at some point (might have been lost in one of many rebases...). Would that be acceptable to you?

```suggestion
    support: Optional[SupportLevel] = None,
    queryable: Optional[SupportLevel] = None,
```